packages/vm-derive/Cargo.toml

```diff
@@ -0,0 +1,8 @@
+[package]
+name = "cosmwasm-vm-derive"
```

Comment:
It's just a loose thought, but maybe we could avoid creating a new crate just for this one macro. Instead, we could add this implementation to `cw-derive` and use it in `cw-vm`. Since `cw-derive` is already a dependency in `cw-std`, there shouldn't be any additional compilation overhead.

Also the `cw-vm-derive-impl` could be moved to `cw-vm`

Reply:
The overhead I was thinking about was building the `blake3` crate in Wasm, since including it in the `derive` package would also include it into the `std` crate (even though we only need it in the VM).

And we have to keep the `impl` separate since it's the actual proc-macro implementation. We could remove the indirection and just make `vm-derive` a proc-macro crate and add `inventory` as a dependency to `vm` directly and include the little bit of code directly in the VM package, but I feel like the current structure is a little cleaner.

Instead of depending on the fact that the package somehow magically has the dependencies we need, we explicitly provide them via the `vm-derive` package as reexports.

Reply:
Okay, now that we got rid of the dependency on `inventory` I personally would have _less_ of a problem adding it to the existing `derive` crate.

The only thing I still would dislike about the approach is that we would add a dependence on the `blake3` crate to every contract out there, meaning they have to compile `blake3` just to use the `entry_point` macro.

It wouldn't be a _huge_ deal tbh, but it's still something that might not be the most optimal solution. Personally, I don't see a problem with adding the `vm-derive` crate, it will just work.

The only thing we have to keep in mind is that we transfer ownership of the crate to the GitHub team, otherwise we will run into a similar problem as we did with `cosmwasm-core` a few months back.

Reply:
Okay, that sounds like a good reason to keep them separate. Let's keep contract compilation lean.